File: .yalc/@entur/atb-mobile-client-sdk/challenge/bluetooth/model/bluetooth-model-ios-lib/Sources/bluetooth-model-ios-lib/header/ChallengeAdvertisingFrameHeader.swift

import Foundation

public final class ChallengeAdvertisingFrameHeader : ChallengeAdvertisingFrameGroupHeader {
    
    static func comparator(a: ChallengeAdvertisingFrameHeader, b: ChallengeAdvertisingFrameHeader) -> Bool {
        if(a.channel != b.channel) {
            return a.channel < b.channel
        }
        
        if(a.clientId != b.clientId) {
            return a.clientId < b.clientId
        }
        
        if(a.correlationId != b.correlationId) {
            return a.correlationId < b.correlationId
        }

        return a.frameNumber < b.frameNumber;
    }
    
    let frameNumber : UInt32
    let lastFrame : Bool
    
    public override var description: String {
        return "clientId: \(clientId) correlationId: \(correlationId) channel: \(channel) frameNumber: \(frameNumber) lastFrame: \(lastFrame)"
    }
    
    init(clientId: UInt32, frameNumber : UInt32, lastFrame : Bool, correlationId: UInt32, channel : UInt32) {
        self.frameNumber = frameNumber;
        self.lastFrame = lastFrame;
        super.init(clientId: clientId, correlationId: correlationId, channel: channel)
    }
    
    public override func hash(into hasher: inout Hasher) {
        hasher.combine(clientId)
        hasher.combine(correlationId)
        hasher.combine(channel)
        hasher.combine(frameNumber)
        hasher.combine(lastFrame)
    }

    public static func == (a:ChallengeAdvertisingFrameHeader, b:ChallengeAdvertisingFrameHeader) -> Bool {
        return
            a.clientId == b.clientId &&
            a.correlationId == b.correlationId &&
            a.channel == b.channel &&
            a.frameNumber == b.frameNumber &&
            a.lastFrame == b.lastFrame
    }
    
    static func < (lhs: ChallengeAdvertisingFrameHeader, rhs: ChallengeAdvertisingFrameHeader) -> Bool {
        return ChallengeAdvertisingFrameHeader.comparator(a: lhs, b: rhs)
    }
    
}

File: .yalc/@entur/atb-mobile-client-sdk/challenge/bluetooth/model/bluetooth-model-java-lib/src/main/java/no/entur/abt/android/bluetooth/header/ChallengeAdvertisingFrameHeader.java
package no.entur.abt.android.bluetooth.header;

import java.util.Comparator;
import java.util.Objects;

public class ChallengeAdvertisingFrameHeader extends ChallengeAdvertisingFrameGroupHeader {

	public static final Comparator<ChallengeAdvertisingFrameHeader> comparator = (a, b) -> {
		int result = Integer.compare(a.getChannel(), b.getChannel());
		if (result != 0) {
			return result;
		}
		result = Integer.compare(a.getClientId(), b.getClientId());
		if (result != 0) {
			return result;
		}
		result = Integer.compare(a.getCorrelationId(), b.getCorrelationId());
		if (result != 0) {
			return result;
		}
		return Integer.compare(a.getFrameNumber(), b.getFrameNumber());
	};

	private final int frameNumber; // 0-based index
	private final boolean lastFrame;

	public ChallengeAdvertisingFrameHeader(int clientId, int frameNumber, boolean lastFrame, int correlationId, int channel) {
		super(clientId, correlationId, channel, Objects.hash(clientId, correlationId, channel, frameNumber, lastFrame));

		this.frameNumber = frameNumber;
		this.lastFrame = lastFrame;
	}

	// i.e. frame index
	public int getFrameNumber() {
		return frameNumber;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o)
			return true;
		if (o == null || getClass() != o.getClass())
			return false;
		ChallengeAdvertisingFrameHeader that = (ChallengeAdvertisingFrameHeader) o;
		return clientId == that.clientId && correlationId == that.correlationId && frameNumber == that.frameNumber && channel == that.channel
				&& lastFrame == that.lastFrame;
	}

	@Override
	public int hashCode() {
		return hash;
	}

	public boolean isLast() {
		return lastFrame;
	}

}
